### PR TITLE
examples: add TLS support to basic price oracle service

### PR DESCRIPTION
**Pull request by ffranr**
This commit adds basic TLS support to the example price oracle service. When connecting to the RPC price oracle, Tapd now assumes TLS. TLS certificate authentication is skipped by Tapd.

Simplify integration of the example price oracle with Tapd. (Hopefully helpful for @HannahMR )

### CODE REVIEW

File: docs/examples/basic-price-oracle/main.go
Comment (gijswijs) [inline docs/examples/basic-price-oracle/main.go:253]: Shouldn't we handle a returned error?
Also, why not use `backendService.GracefulStop()`?

**Review by gijswijs — APPROVED** [review]
LGTM! :tada:
I can still see some magic numbers in there, but that's expected for essentially a mock. But I just wanted to flag it.

Also, this server now _only_ works with TLS. That's what we want, right?

Made a nit comment, but nothing that would block merging.

**Comment by ffranr** [comment]
> Also, this server now only works with TLS. That's what we want, right?

@gijswijs yes, I think it will do for now. And tapd expects that right now.

**Review by gijswijs — APPROVED** [review]

**Comment by HannahMR** [comment]
This does solve the communication error between litd and the oracle in my testing.

**Comment by ffranr** [comment]
> Also, this server now only works with TLS. That's what we want, right?

That's fine IMO. We just want to make sure that it can work with tapd out of the box.

File: docs/examples/basic-price-oracle/main.go
Comment (dstadulis) [inline docs/examples/basic-price-oracle/main.go:244]: What's the rational for expiring the cert in 24 hours?

File: docs/examples/basic-price-oracle/main.go
Comment (dstadulis) [inline docs/examples/basic-price-oracle/main.go:231]: Would other curves be more desired? p256 vs https://pkg.go.dev/crypto/elliptic#P521

**Comment by ffranr** [comment]
I've added logging to this and the format is sub-optimal. So I will improve before merge.

File: docs/examples/basic-price-oracle/main.go
Comment (ffranr) [reply docs/examples/basic-price-oracle/main.go:244]: Just a reasonable number that won't get in the way of running the basic example.

File: docs/examples/basic-price-oracle/main.go
Comment (ffranr) [reply docs/examples/basic-price-oracle/main.go:231]: This is just to get TLS working with the example service. Its just example code and not security critical.

**Pull request merged**
